jupyterhub/apihandlers/users.py

```diff
@@ -150,6 +150,12 @@ def patch(self, name):
             raise web.HTTPError(404)
         data = self.get_json_body()
         self._check_user_model(data)
+        # check if the new name is already taken
+        new_name = self.find_user(data['name']) if 'name' in data else None
+        if new_name is not None:
```

Comment:
but you have to put another if under if.self to check if name and data['name'] are the same.....